<launch>
    <arg name="robot_1" value="robot_1" />
    <arg name="robot_2" value="robot_2" />

    <arg name="environment" value="office" />
    <arg name="metric_map" default="$(find learning_toponav)/maps/$(arg environment)/metric.yaml" />
    <arg name="topomap" default="$(find learning_toponav)/maps/$(arg environment)/topomap.tpg" />
    <arg name="topics_yaml" default="$(find learning_toponav)/param/topics.yaml" />
    <arg name="rviz_config" default="-d $(find learning_toponav)/rviz/multi_toponav.rviz" />
    <arg name="gazebo_gui" default="false" /> <!-- ACTIVATE GAZEBO GUI -->
    <arg name="rqt_console" default="false" /> <!-- LAUNCH RQT CONSOLE -->

    <!-- GAZEBO -->
    <include file="$(find learning_multirobot)/launch/includes/main_gazebo.launch">
        <arg name="world_name" value="$(find learning_multirobot)/worlds/$(arg environment).world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="$(arg gazebo_gui)"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <!--gzserver killer-->
	<node pkg="my_fun" type="gz_killer.py" name="gz_killer" output="screen" />

    <!-- map server -->
    <node pkg="map_server" type="map_server" name="map_loader" args="$(arg metric_map)" />

    <!--toponodes publisher-->
    <node pkg="learning_toponav" type="toponodes_publisher.py" name="toponodes_publisher"
          args="--topomap $(arg topomap) --yaml $(arg topics_yaml)"
    />

    <!--robot 1-->
    <group ns="$(arg robot_1)">
        <include file="$(find learning_toponav)/launch/includes/one_robot.launch">
            <arg name="robot_name" value="$(arg robot_1)" />
            <arg name="color" value="red" />
            <arg name="topics_yaml" value="$(arg topics_yaml)" />

            <arg name="x_pos" value="1" />
            <arg name="y_pos" value="1" />
            <arg name="z_pos" value="0" />
        </include>
    </group>

    <!--robot 2-->
    <group ns="$(arg robot_2)">
        <include file="$(find learning_toponav)/launch/includes/one_robot.launch">
            <arg name="robot_name" value="$(arg robot_2)" />
            <arg name="color" value="blue" />
            <arg name="topics_yaml" value="$(arg topics_yaml)" />

            <arg name="x_pos" value="-1" />
            <arg name="y_pos" value="1" />
            <arg name="z_pos" value="0" />
        </include>
    </group>

    <!--topoplanner-->
    <arg name="adjlist" value="$(find learning_toponav)/maps/$(arg environment)/adjlist.txt" />
    <node pkg="learning_toponav" type="topoplanner.py" name="topoplanner"
          args="--adjlist $(arg adjlist) --environment $(arg environment) --yaml $(arg topics_yaml)"
          output="screen"
    />

    <!--destinations debugger-->
    <arg name="dest_debug" default="true" />
    <node pkg="learning_toponav" type="destinations_debugger.py" name="destinations_debugger"
          args="--yaml $(find learning_toponav)/param/topics.yaml" if="$(arg dest_debug)"
    />

    <!-- RVIZ -->
    <node name="rviz" pkg="rviz" type="rviz" args="$(arg rviz_config)" />

    <!--goal dispatcher-->
    <!--<node pkg="learning_multirobot" type="simple_goal_publisher.py" name="goal_publisher"-->
    <!--      args="&#45;&#45;yaml $(arg topics_yaml)" output="screen"-->
    <!--/>-->

    <!--rqt_console-->
    <node pkg="rqt_console" type="rqt_console" name="rqt_console" if="$(arg rqt_console)" />
</launch>